### PR TITLE
[FLINK-27929][python] Drop support for python36

### PR DESCRIPTION
**Pull request by bzhaoopenstack**
This PR will drop the py36 support for pyflink.

## What is the purpose of the change

py36 had reached its EOL. https://endoflife.date/python
So need to drop it from pyflink.

## Brief change log

  - drop system py36 version validation, increase to 3.7
  - associate py36 doc change

## Verifying this change
py36 won't be supported by pyflink 1.17

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-python/setup.py [hunk 1]
```diff
@@ -27,12 +27,12 @@
...
 from setuptools import setup, Extension
 
-if sys.version_info < (3, 6):
-    print("Python versions prior to 3.6 are not supported for PyFlink.",
+if sys.version_info < (3, 7):
+    print("Python versions prior to 3.7 are not supported for PyFlink.",
           file=sys.stderr)
     sys.exit(-1)
-elif sys.version_info.minor == 6:
-    warnings.warn("Python version 3.6 won't be supported for PyFlink after 1.16.")
```
Comment (HuangXingBo) [inline flink-python/setup.py:35]: we haven't planned to drop support python3.7

File: flink-python/setup.py [hunk 2]
```diff
@@ -316,16 +316,10 @@ def extracted_output_files(base_dir, file_path, output_directory):
...
-        # python 3.6 upper and lower limit
-        install_requires.append('numpy>=1.14.3,<1.20')
-        install_requires.append('pandas>=1.0,<1.2.0')
-        install_requires.append('pyarrow>=0.15.1,<7.0.0')
-    else:
-        # python 3.7, 3.8 and 3.9 upper limit and M1 chip lower limit,
-        install_requires.append('numpy>=1.21.4,<1.22.0')
-        install_requires.append('pandas>=1.3.0,<1.4.0')
-        install_requires.append('pyarrow>=5.0.0,<9.0.0')
+    # python 3.7, 3.8 and 3.9 upper limit and M1 chip lower limit,
```
Comment (HuangXingBo) [inline flink-python/setup.py:319]: since we drop the support python3.6, we can put these install dependencies in `install_requires` directly.

**Review by HuangXingBo — COMMENTED** [review]
@bzhaoopenstack Thanks a lot for the PR. I left some minor comments and we need to update relevant comment of python support version in docs too.

**Comment by dianfu** [comment]
Closed this PR as this should have already been supported.

**Pull request closed without merging**